hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -347,4 +351,12 @@ private static void inferAvroSchema(Configuration conf, LogicalType rowType) {
       conf.setString(FlinkOptions.SOURCE_AVRO_SCHEMA, inferredSchema);
     }
   }
+
+  private static void setupRootOptions(Configuration conf, ReadableConfig configuration) {
+    if (configuration instanceof TableConfig) {
+      ((Configuration)((TableConfig) configuration).getRootConfiguration()).toMap().forEach((rootConfigKey, rootConfigValue) -> {
+        conf.setString(rootConfigKey, rootConfigValue);
```

Comment:
Not sure what kind of options are passed through the `context.getConfiguration()`, seems the options from `TableConfig` ? Actually if you use the SQL, the option with prefix `hadoop.` can be passed correctly.

Reply:
Hi @danny0405 ,
Thanks for looking at this.

The issue is when we try to use Hudi with Flink and Azure.
The flink configurations includes properties which are needed by Hadoop to connect to Azure
In the flow from Flink -> Hudi -> Hadoop -> Storage for table, the configurations are not passed.

The hadoop-azure library never gets the configs for ClientId, Credentials, AuthType, etc.
Made this change to pass these configuration from Flink Library to Hadoop library.

Another work around is to send the configs as part of the CatalogTable Options. but that does not seem correct.
Let me know your thoughts though.

Reply:
The `TableConfig` is used for passing around Flink SQL job config options, not very suitable for hadoop configurations.

Can you just put the specific hadoop config options into the hadoop `core-site.xml` file  which stays under the dir `HADOOP_CONF_DIR` `HADOOP_HOME/conf` `HADOOP_HOME/etc/hadoop` ?

Or just use the table options which are supported now ~

Reply:
Sure @danny0405 .
Can I close the Merge request then ? 
Can we add these to the Hudi documentation ?